kvbc/src/Replica.cpp

```diff
@@ -323,8 +323,25 @@ void Replica::createReplicaAndSyncState() {
   handleNewEpochEvent();
   handleWedgeEvent();
   bftEngine::impl::DbCheckpointManager::instance().initializeDbCheckpointManager(
-      m_dbSet.dataDBClient, m_replicaPtr->persistentStorage(), aggregator_, [this]() -> uint64_t {
-        return getLastBlockId();
+      m_dbSet.dataDBClient,
+      m_replicaPtr->persistentStorage(),
+      aggregator_,
+      [this]() -> uint64_t { return getLastBlockId(); },
+      [](BlockId blockIdAtCheckpoint, const std::string &path) {
+        if (blockIdAtCheckpoint < INITIAL_GENESIS_BLOCK_ID) {
```

Comment:
The logic of checking the block id is given in deleteLastReachableBlock() function. If the block id is wrong then the function is throwing logic_error exception. It's better to catch the exception and return, instead of rechecking all the conditions before calling the function.

Reply:
You are right, but we are checking the input only here, not any current kvbc state. Does that make sense?